In [1]:
%run _standard_imports.ipynb

python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
numpy 1.11.2
scipy 0.18.1
pandas 0.19.0
numexpr 2.6.1
pysam 0.8.4
petl 1.1.0
petlx 1.0.3
vcf 0.6.8
vcfnp 2.2.0
h5py 2.6.0
tables 3.3.0


In [108]:
panoptes_metadata_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_panoptes_samples_20170124.txt.gz"
panoptes_final_metadata_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/pf_6_0_panoptes_samples_final_20170124.txt.gz"
oxford_table_fn = "/nfs/team112_internal/rp7/data/methods-dev/builds/Pf6.0/20161214_samples_which_release/oxford.txt"
pf3k_metadata_fn = "/nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/pf3k_release_5_metadata.txt"
crosses_metadata_fn = "/nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/pf3k_release_5_crosses_metadata.txt"
new_pf3k_metadata_fn = "/nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/pf3k_release_5_metadata_20170210.txt.gz"

In [4]:
tbl_pf_60_metadata = (
    etl
    .fromtsv(panoptes_metadata_fn)
#     .cut(['Sample', 'pf60_OxfordSrcCode', 'pf60_OxfordDonorCode', 'Individual_ID', 'pf60_AlfrescoStudyCode',
#           'pf60_HasDuplicate', 'pf60_DiscardAsDuplicate', 'pc_pass_missing', 'pc_genome_covered_at_1x'])
)
print(len(tbl_pf_60_metadata.data()))
tbl_pf_60_metadata

7182


+--------+------------+---------------+-----------------+---------------------+-------+-------+--------------+--------+--------+------------------+------------------+------------+-------------+-------------+-----------+------+------+--------------+--------------------+------------------+-------------------------------------------+---------------+---------------+-----------------------+--------------+---------------+--------------------+------------+--------------------+----------------------+------------------------+-----------+-------------+-----------+-----------+------------------+------------------+---------------+-------------------+-------------------------+-----------------------+-------------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-------------------------------+--------------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+---------------------------+------------------------+------------+------------+---------------------------+---------------------------+----------------------------+----------------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-------------------+-----------+------------------+-----------+-----------------------+-----------------------+---------------+-----------+-----------------+
| Num    | Sample     | OxfordSrcCode | OxfordDonorCode | AlfrescoStudyCode   | Study | Type  | NotAmplified | InV5   | InV4   | SubContinentCode | SubContinentName | RegionCode | CountryCode | CountryName | Location  | Year | Date | HasDuplicate | DiscardAsDuplicate | ManuallyExcluded | run_accessions                            | sampling_date | sampling_Year | arrival_date          | arrival_Year | batch_arrived | batch_arrived_year | elife_Year | pf60_OxfordSrcCode | pf60_OxfordDonorCode | pf60_AlfrescoStudyCode | pf60_Type | pf60_InV5.1 | pf60_InV5 | pf60_InV4 | pf60_CountryCode | pf60_CountryName | Individual_ID | pf60_HasDuplicate | pf60_DiscardAsDuplicate | pc_pass_missing       | pc_genome_covered_at_1x |

In [5]:
(
    tbl_pf_60_metadata
    .valuecounts('AlfrescoStudyCode')
    .displayall()
)

AlfrescoStudyCode 
 count 
 frequency 
 
 
 
 
 1052-PF-TRAC-WHITE 
 1171 
 0.1630465051517683 
 
 
 1125-PF-TH-NOSTEN 
 671 
 0.09342801448064605 
 
 
 1044-PF-KH-FAIRHURST 
 504 
 0.07017543859649122 
 
 
 1017-PF-GH-AMENGA-ETEGO 
 390 
 0.05430242272347535 
 
 
 
 381 
 0.05304928989139515 
 
 
 1022-PF-MW-OCHOLLA 
 349 
 0.04859370648844333 
 
 
 1095-PF-TZ-ISHENGOMA 
 300 
 0.04177109440267335 
 
 
 1094-PF-GH-AMENGA-ETEGO 
 255 
 0.035505430242272346 
 
 
 1093-PF-CM-APINJOH 
 239 
 0.033277638540796434 
 
 
 1008-PF-SEA-RINGWALD 
 234 
 0.03258145363408521 
 
 
 1026-PF-GN-CONWAY 
 191 
 0.026594263436368698 
 
 
 1031-PF-SEA-PLOWE 
 177 
 0.024644945697577275 
 
 
 1101-PF-CD-ONYAMBOKO 
 174 
 0.02422723475355054 
 
 
 1127-PF-ML-SOULEYMANE 
 164 
 0.022834864940128097 
 
 
 1097-PF-ML-MAIGA 
 137 
 0.019075466443887495 
 
 
 1010-PF-TH-ANDERSON 
 107 
 0.01489835700362016 
 
 
 1131-PF-BJ-BERTIN 
 102 
 0.014202172096908938 
 
 
 1096-PF-GH-GHANSAH 
 101 
 0.014062935115566694 
 
 
 1083-PF-GH-CONWAY 
 100 
 0.01392369813422445 
 
 
 1136-PF-GM-NGWA 
 100 
 0.01392369813422445 
 
 
 1001-PF-ML-DJIMDE 
 96 
 0.013366750208855471 
 
 
 1147-PF-MR-CONWAY 
 86 
 0.011974380395433026 
 
 
 1062-PF-PG-BARRY 
 82 
 0.011417432470064048 
 
 
 1006-PF-GM-CONWAY 
 79 
 0.010999721526037315 
 
 
 1138-PF-CD-FANELLO 
 77 
 0.010721247563352826 
 
 
 1100-PF-CI-YAVO 
 70 
 0.009746588693957114 
 
 
 1155-PF-ID-PRICE 
 70 
 0.009746588693957114 
 
 
 1137-PF-GM-DALESSANDRO 
 68 
 0.009468114731272625 
 
 
 1107-PF-KEN-KAMAU 
 61 
 0.008493455861876914 
 
 
 1135-PF-SN-CONWAY 
 60 
 0.00835421888053467 
 
 
 1015-PF-KE-NZILA 
 59 
 0.008214981899192425 
 
 
 1021-PF-PG-MUELLER 
 56 
 0.007797270955165692 
 
 
 1004-PF-BF-OUEDRAOGO 
 56 
 0.007797270955165692 
 
 
 1134-PF-ML-CONWAY 
 52 
 0.007240323029796714 
 
 
 1007-PF-TZ-DUFFY 
 50 
 0.006961849067112225 
 
 
 1011-PF-KH-SU 
 40 
 0.00556947925368978 
 
 
 1103-PF-PDN-GMSN-NGWA 
 34 
 0.004734057365636313 
 
 
 1098-PF-ET-GOLASSA 
 34 
 0.004734057365636313 
 
 
 1102-PF-MG-RANDRIANARIVELOJOSIA 
 25 
 0.0034809245335561123 
 
 
 1020-PF-VN-BONI 
 24 
 0.003341687552213868 
 
 
 1106-PV-MULTI-PRICE 
 22 
 0.003063213589529379 
 
 
 1016-PF-TH-NOSTEN 
 21 
 0.0029239766081871343 
 
 
 1023-PF-CO-ECHEVERRI-GARCIA 
 17 
 0.0023670286828181564 
 
 
 1013-PF-PEGB-BRANCH 
 16 
 0.002227791701475912 
 
 
 1133-PF-LAB-MERRICK 
 14 
 0.001949317738791423 
 
 
 1024-PF-UG-BOUSEMA 
 14 
 0.001949317738791423 
 
 
 1145-PF-PE-GAMBOA 
 13 
 0.0018100807574491784 
 
 
 1027-PF-KE-BULL 
 11 
 0.0015316067947646894 
 
 
 1014-PF-SSA-SUTHERLAND 
 10 
 0.001392369813422445 
 
 
 1025-PF-KH-PLOWE 
 10 
 0.001392369813422445 
 
 
 1009-PF-KH-PLOWE 
 5 
 0.0006961849067112225 
 
 
 1012-PF-KH-WHITE 
 2 
 0.000278473962684489 
 
 
 1018-PF-GB-NEWBOLD 
 1 
 0.0001392369813422445

In [6]:
(
    tbl_pf_60_metadata
    .selectin('AlfrescoStudyCode', ['1155-PF-ID-PRICE', '1106-PV-MULTI-PRICE'])
    .valuecounts('pf60_Location')
    .displayall()
)

pf60_Location 
 count 
 frequency 
 
 
 
 
 Papua 
 92 
 1.0

In [96]:
tbl_pf_60_metadata = (
    etl
    .fromtsv(panoptes_final_metadata_fn)
#     .cut(['Sample', 'OxfordSrcCode', 'OxfordDonorCode', 'Individual_ID', 'IndividualGroup', 'AlfrescoStudyCode',
#           'HasDuplicate', 'DiscardAsDuplicate', 'pc_pass_missing', 'pc_genome_covered_at_1x'])
    .cut(['Sample', 'OxfordSrcCode', 'OxfordDonorCode', 'Individual_ID'])
)
print(len(tbl_pf_60_metadata.data()))
tbl_pf_60_metadata

7182


+------------+---------------+-----------------+---------------+
| Sample     | OxfordSrcCode | OxfordDonorCode | Individual_ID |
+============+===============+=================+===============+
| 'PM0001-C' | '215C'        | '215C'          | '215C'        |
+------------+---------------+-----------------+---------------+
| 'PM0002-C' | '217C'        | '217C'          | '217C'        |
+------------+---------------+-----------------+---------------+
| 'PM0003-C' | '520C'        | '520C'          | '520C'        |
+------------+---------------+-----------------+---------------+
| 'PM0004-C' | 'A336TPI7C'   | 'A336TPI7C'     | 'A336TPI7C'   |
+------------+---------------+-----------------+---------------+
| 'PM0103-C' | 'ALF003'      | 'ALF003'        | 'ALF003'      |
+------------+---------------+-----------------+---------------+
...

In [8]:
tbl_pf3k_metadata = (
    etl
    .fromtsv(pf3k_metadata_fn)
)
print(len(tbl_pf3k_metadata.data()))
tbl_pf3k_metadata

2640


+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+
| sample     | acc         | study  | study_title                                               | contact_name          | contact_email  | country      | site         | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable |
+============+=============+========+===========================================================+=======================+================+==============+==============+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+
| 'PA0007-C' | 'ERS010116' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '2084040899' | '2002872367' | '859315087'      | '75'            | '23.2'            | '22925304'           | '20607683'           | '17717393'            | '9323097'             | '77.03'       | '328.9'            | '144.7'          | '76.94'   |
+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+
| 'PA0008-C' | 'ERS010126' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '2540709087' | '2467552323' | '461921509'      | '75'            | '23.9'            | '23280178'           | '23094690'           | '22558802'            | '15064542'            | '101.12'      | '343.3'            | '72.5'           | '89.33'   |
+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+
| 'PA0011-C' | 'ERS010043' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '3719467019' | '3587824835' | '173365999'      | '75'            | '25.8'            | '23129884'           | '23029563'           | '22958094'            | '22173814'            | '153.9'       | '249.7'            | '45'             | '89.56'   |
+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+---

In [9]:
tbl_oxford = (
    etl
    .fromtsv(oxford_table_fn)
    .distinct('oxford_code')
    .cut(['oxford_code', 'oxford_source_code', 'oxford_donor_source_code'])
)
print(len(tbl_oxford.distinct('oxford_code').data()))
print(len(tbl_oxford.data()))
tbl_oxford

210873
210873


+-------------+--------------------+--------------------------+
| oxford_code | oxford_source_code | oxford_donor_source_code |
+=============+====================+==========================+
| 'AA0001-C'  | 'Ghana_18-F6'      | 'Ghana_18-F6'            |
+-------------+--------------------+--------------------------+
| 'AA0002-C'  | 'Ghana_18-G1'      | 'Ghana_18-G1'            |
+-------------+--------------------+--------------------------+
| 'AA0003-C'  | '3-C7'             | '3-C7'                   |
+-------------+--------------------+--------------------------+
| 'AA0003-Cx' | '3-C7'             | '3-C7'                   |
+-------------+--------------------+--------------------------+
| 'AA0004-C'  | '3-B10'            | '3-B10'                  |
+-------------+--------------------+--------------------------+
...

In [18]:
tbl_crosses = (
    etl
    .fromtsv(crosses_metadata_fn)
    .cut(['sample', 'study', 'clone'])
)
print(len(tbl_crosses.distinct('sample').data()))
print(len(tbl_crosses.data()))
tbl_crosses

98
98


+-------------+--------+--------+
| sample      | study  | clone  |
+=============+========+========+
| 'PG0008-CW' | '1041' | 'DD2'  |
+-------------+--------+--------+
| 'PG0004-CW' | '1041' | 'HB3'  |
+-------------+--------+--------+
| 'PG0023-C'  | '1041' | '1BB5' |
+-------------+--------+--------+
| 'PG0022-Cx' | '1041' | '3BA6' |
+-------------+--------+--------+
| 'PG0024-C'  | '1041' | '3BD5' |
+-------------+--------+--------+
...

In [31]:
def is_duplicate(prv, cur, nxt):
    if prv is None and (nxt['Individual_ID'] == cur['Individual_ID']):
        return(True)
    elif prv is None and (nxt['Individual_ID'] != cur['Individual_ID']):
        return(False)
    elif nxt is None and (prv['Individual_ID'] == cur['Individual_ID']):
        return(True)
    elif nxt is None and (prv['Individual_ID'] != cur['Individual_ID']):
        return(False)
    elif(
            (prv['Individual_ID'] == cur['Individual_ID']) or
            (nxt['Individual_ID'] == cur['Individual_ID'])
        ):
        return(True)
    else:
        return(False)
    
def discard_as_duplicate(prv, cur, nxt):
    if prv is None:
        return(False)
    elif (prv['Individual_ID'] == cur['Individual_ID']):
        return(True)
    else:
        return(False)
    



In [120]:
# final_columns = list(tbl_pf3k_metadata.header()) + ['IsFieldSample', 'PreferredSample', 'AllSamplesThisIndividual', 'DiscardAsDuplicate', 'HasDuplicate', 'Individual_ID']
final_columns = list(tbl_pf3k_metadata.header()) + ['IsFieldSample', 'PreferredSample', 'AllSamplesThisIndividual']
final_columns

['sample',
 'acc',
 'study',
 'study_title',
 'contact_name',
 'contact_email',
 'country',
 'site',
 'collection_date',
 'sample_prep',
 'bases',
 'bases_mapped',
 'bases_duplicated',
 'avg_read_length',
 'mean_base_quality',
 'bases_of_1X_coverage',
 'bases_of_5X_coverage',
 'bases_of_10X_coverage',
 'bases_of_50X_coverage',
 'mean_coverage',
 'mean_fragment_size',
 'sd_fragment_size',
 '%callable',
 'IsFieldSample',
 'PreferredSample',
 'AllSamplesThisIndividual']

In [121]:
tbl_temp = (
    tbl_pf3k_metadata
    .leftjoin(tbl_pf_60_metadata, lkey='sample', rkey='Sample')
    .leftjoin(tbl_oxford, lkey='sample', rkey='oxford_code')
    .leftjoin(tbl_crosses, key='sample', rprefix='crosses_')
    .convert('Individual_ID', lambda v, r: r['crosses_clone'], where=lambda r: r['study'] in ['1041', '1042', '1043'], pass_row=True)
    .convert('Individual_ID', lambda v, r: r['sample'], where=lambda r: r['study'] in ['Broad Senegal', '1104', ''], pass_row=True)
    .convert('Individual_ID', lambda v, r: r['oxford_donor_source_code'], where=lambda r: r['Individual_ID'] is None, pass_row=True)
    .convert('Individual_ID', lambda v, r: 'PF955', where=lambda r: r['Individual_ID'] == 'PF955_MACS', pass_row=True)
    .addfield('IsFieldSample', lambda r: r['country'] != '')
    .sort(['Individual_ID', 'bases_of_5X_coverage'], reverse=True)
    .addfieldusingcontext('HasDuplicate', is_duplicate)
    .addfieldusingcontext('DiscardAsDuplicate', discard_as_duplicate)
    .addfield('PreferredSample', lambda rec: rec['DiscardAsDuplicate'] == False)
)

# tbl_temp.totsv('temp.txt')
# tbl_temp = etl.fromtsv('temp.txt')

tbl_duplicates_sample = (
    tbl_temp
    .aggregate('Individual_ID', etl.strjoin(','), 'sample')
    .rename('value', 'AllSamplesThisIndividual')
)

tbl_new_pf3k_metadata = (
    tbl_temp
    .leftjoin(tbl_duplicates_sample, key='Individual_ID')
    .cut(final_columns)
    .sort('sample')
    .sort('IsFieldSample', reverse=True)
)

print(len(tbl_new_pf3k_metadata.distinct('sample').data()))
print(len(tbl_new_pf3k_metadata.data()))
tbl_new_pf3k_metadata

2640
2640


+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+
| sample     | acc         | study  | study_title                                               | contact_name          | contact_email  | country      | site         | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual |
+============+=============+========+===========================================================+=======================+================+==============+==============+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+
| 'PA0007-C' | 'ERS010116' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '2084040899' | '2002872367' | '859315087'      | '75'            | '23.2'            | '22925304'           | '20607683'           | '17717393'            | '9323097'             | '77.03'       | '328.9'            | '144.7'          | '76.94'   | True          | True            | 'PA0007-C'               |
+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+
| 'PA0008-C' | 'ERS010126' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '2540709087' | '2467552323' | '461921509'      | '75'            | '23.9'            | '23280178'           | '23094690'           | '22558802'            | '15064542'            | '101.12'      | '343.3'            | '72.5'           | '89.33'   | True          | True            | 'PA0008-C'               |
+------------+-------------+--------+-----------------------------------------------------------+-----------------------+----------------+--------------+--------------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+
| 'PA0011-C' | 'ERS010043' | '1006' | 'Genome-wide analysis of genetic variation in the Gambia' | 'Alfred Amambua-Ngwa' | 'angwa@mrc.gm' | 'The Gambia' | 'GM_Coastal' | ''              | ''          | '3719467019' | '3587824835' | '173365999'      | '75'            | '25.8'            | '23129884'           | '23029563'           | '22958094'            | '22173814'            | '153.9'       | '249.7'            | '45'

In [122]:
tbl_new_pf3k_metadata.totsv(new_pf3k_metadata_fn, lineterminator='\n')
# tbl_new_pf3k_metadata.totsv(new_pf3k_metadata_fn)

In [123]:
# Sanity check new file is same as old but with extra_columns
!zcat {new_pf3k_metadata_fn} | cut -f 1-23 > /nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_1.txt

In [124]:
!sed 's/"//g' /nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/pf3k_release_5_metadata.txt > \
/nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_2.txt

In [125]:
!diff /nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_1.txt \
/nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_2.txt

In [126]:
!rm /nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_1.txt
!rm /nfs/team112_internal/production/release_build/Pf3K/pilot_5_0/temp_remove_2.txt


# Sanity checks

In [106]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'PF955')

+------------+-------------+--------+----------------------------------------------+------------------+----------------------------+------------+----------+-----------------+-------------+---------------+---------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| sample     | acc         | study  | study_title                                  | contact_name     | contact_email              | country    | site     | collection_date | sample_prep | bases         | bases_mapped  | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual | DiscardAsDuplicate | HasDuplicate | Individual_ID |
+============+=============+========+==============================================+==================+============================+============+==========+=================+=============+===============+===============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+====================+==============+===============+
| 'PH0294-C' | 'ERS025100' | '1044' | 'Genomics of parasite clearance in Cambodia' | 'Rick Fairhurst' | 'RFairhurst@niaid.nih.gov' | 'Cambodia' | 'Pursat' | ''              | ''          | '14376785559' | '14017971684' | '844158876'      | '74'            | '25.3'            | '23232066'           | '23184175'           | '23160442'            | '23045365'            | '560.91'      | '283'              | '43.4'           | '90.11'   | True          | True            | 'PH0294-C'               | False              | False        | 'PF955'       |
+------------+-------------+--------+----------------------------------------------+------------------+----------------------------+------------+----------+-----------------+-------------+---------------+---------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+

In [90]:
tbl_new_pf3k_metadata.valuecounts('IsFieldSample')

+---------------+-------+----------------------+
| IsFieldSample | count | frequency            |
+===============+=======+======================+
| 'True'        |  2512 |   0.9515151515151515 |
+---------------+-------+----------------------+
| 'False'       |   128 | 0.048484848484848485 |
+---------------+-------+----------------------+

In [102]:
tbl_new_pf3k_metadata.valuecounts('PreferredSample')

+-----------------+-------+----------------------+
| PreferredSample | count | frequency            |
+=================+=======+======================+
| True            |  2513 |   0.9518939393939394 |
+-----------------+-------+----------------------+
| False           |   127 | 0.048106060606060604 |
+-----------------+-------+----------------------+

In [103]:
tbl_new_pf3k_metadata.valuecounts('DiscardAsDuplicate')

+--------------------+-------+----------------------+
| DiscardAsDuplicate | count | frequency            |
+====================+=======+======================+
| False              |  2513 |   0.9518939393939394 |
+--------------------+-------+----------------------+
| True               |   127 | 0.048106060606060604 |
+--------------------+-------+----------------------+

In [104]:
tbl_new_pf3k_metadata.selecteq('PreferredSample', False).displayall()

sample 
 acc 
 study 
 study_title 
 contact_name 
 contact_email 
 country 
 site 
 collection_date 
 sample_prep 
 bases 
 bases_mapped 
 bases_duplicated 
 avg_read_length 
 mean_base_quality 
 bases_of_1X_coverage 
 bases_of_5X_coverage 
 bases_of_10X_coverage 
 bases_of_50X_coverage 
 mean_coverage 
 mean_fragment_size 
 sd_fragment_size 
 %callable 
 IsFieldSample 
 PreferredSample 
 AllSamplesThisIndividual 
 DiscardAsDuplicate 
 HasDuplicate 
 Individual_ID 
 
 
 
 
 PF0550-Cx 
 ERS193635 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 1128003177 
 1080227777 
 296103081 
 99 
 25 
 23193929 
 22559304 
 21328090 
 10623762 
 46.11 
 247.3 
 81 
 87.17 
 True 
 False 
 PF0550-C,PF0550-Cx 
 True 
 True 
 EIMK002 
 
 
 PF0554-Cx 
 ERS193640 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 1610000103 
 1541368403 
 424575393 
 99 
 25.1 
 23173514 
 22787688 
 22091029 
 14398186 
 65.92 
 247.4 
 84 
 88.34 
 True 
 False 
 PF0554-C,PF0554-Cx 
 True 
 True 
 EIMK009 
 
 
 PF0556-Cx 
 ERS193645 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 2205895276 
 2175715476 
 600271069 
 99 
 25.6 
 23126310 
 22747626 
 22166464 
 16533660 
 93.37 
 249.3 
 79.8 
 88.58 
 True 
 False 
 PF0556-C,PF0556-Cx 
 True 
 True 
 EIMK011 
 
 
 PF0557-Cx 
 ERS193650 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 2369153185 
 2336996985 
 638012939 
 99 
 25.3 
 23289320 
 23121311 
 22759386 
 17479969 
 98.99 
 250 
 82.4 
 90.07 
 True 
 False 
 PF0557-C,PF0557-Cx 
 True 
 True 
 EIMK012 
 
 
 PF0559-Cx 
 ERS193655 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 2456917755 
 2422476655 
 666152330 
 99 
 25.5 
 23156246 
 22898036 
 22501037 
 17783127 
 103.79 
 241.7 
 77.5 
 89.2 
 True 
 False 
 PF0559-C,PF0559-Cx 
 True 
 True 
 EIMK015 
 
 
 PF0560-Cx 
 ERS193660 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 1060501745 
 1004195045 
 277725138 
 99 
 25.2 
 23054561 
 22251574 
 20956171 
 9604100 
 43.18 
 234.9 
 76.2 
 86.09 
 True 
 False 
 PF0560-C,PF0560-Cx 
 True 
 True 
 EIMK017 
 
 
 PF0562-Cx 
 ERS193665 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 1033720919 
 982964719 
 273415835 
 99 
 24.9 
 23146635 
 22220473 
 20699777 
 9211730 
 42.01 
 246.5 
 82.2 
 85.44 
 True 
 False 
 PF0562-C,PF0562-Cx 
 True 
 True 
 EIMK020 
 
 
 PF0578-Cx 
 ERS188107 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 3550554879 
 3482093479 
 142276849 
 99 
 25.5 
 23232066 
 23043679 
 22817591 
 19832014 
 147.67 
 246.3 
 71.2 
 90.13 
 True 
 False 
 PF0578-C,PF0578-Cx 
 True 
 True 
 EIMK077 
 
 
 PF0579-Cx 
 ERS193675 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 2260692627 
 2212025427 
 600591176 
 99 
 25.3 
 23288295 
 23106015 
 22681377 
 16988926 
 93.93 
 252 
 81.1 
 90.03 
 True 
 False 
 PF0579-C,PF0579-Cx 
 True 
 True 
 EIMK081 
 
 
 PF0583-Cx 
 ERS188114 

In [107]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'EIMK002').displayall()

sample 
 acc 
 study 
 study_title 
 contact_name 
 contact_email 
 country 
 site 
 collection_date 
 sample_prep 
 bases 
 bases_mapped 
 bases_duplicated 
 avg_read_length 
 mean_base_quality 
 bases_of_1X_coverage 
 bases_of_5X_coverage 
 bases_of_10X_coverage 
 bases_of_50X_coverage 
 mean_coverage 
 mean_fragment_size 
 sd_fragment_size 
 %callable 
 IsFieldSample 
 PreferredSample 
 AllSamplesThisIndividual 
 DiscardAsDuplicate 
 HasDuplicate 
 Individual_ID 
 
 
 
 
 PF0550-C 
 ERS193635 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 01/09/2012 
 
 1687516306 
 1475824006 
 79915784 
 99 
 25.1 
 23264010 
 23152338 
 22920938 
 15665358 
 62.84 
 235.7 
 81.1 
 90.61 
 True 
 True 
 PF0550-C,PF0550-Cx 
 False 
 True 
 EIMK002 
 
 
 PF0550-Cx 
 ERS193635 
 1083 
 Alternative molecular mechanisms for erythrocyte invasion by Plasmodium falciparum in Ghana 
 Gordon Awandare 
 gawandare@hotmail.com 
 Ghana 
 Kintampo 
 28/02/2014 
 
 1128003177 
 1080227777 
 296103081 
 99 
 25 
 23193929 
 22559304 
 21328090 
 10623762 
 46.11 
 247.3 
 81 
 87.17 
 True 
 False 
 PF0550-C,PF0550-Cx 
 True 
 True 
 EIMK002

In [99]:
tbl_temp.valuecounts('DiscardAsDuplicate')

+--------------------+-------+----------------------+
| DiscardAsDuplicate | count | frequency            |
+====================+=======+======================+
| 'False'            |  2513 |   0.9518939393939394 |
+--------------------+-------+----------------------+
| 'True'             |   127 | 0.048106060606060604 |
+--------------------+-------+----------------------+

In [100]:
tbl_new_pf3k_metadata.valuecounts('HasDuplicate')

+--------------+-------+---------------------+
| HasDuplicate | count | frequency           |
+==============+=======+=====================+
| 'False'      |  2388 |  0.9045454545454545 |
+--------------+-------+---------------------+
| 'True'       |   252 | 0.09545454545454546 |
+--------------+-------+---------------------+

In [78]:
tbl_new_pf3k_metadata.selectnone('HasDuplicate').valuecounts('study').displayall()

study 
 count 
 frequency 
 
 
 
 
 Broad Senegal 
 137 
 0.4254658385093168 
 
 
 1043 
 38 
 0.11801242236024845 
 
 
 1041 
 37 
 0.11490683229813664 
 
 
 1104 
 27 
 0.08385093167701864 
 
 
 1042 
 21 
 0.06521739130434782 
 
 
 1022 
 20 
 0.062111801242236024 
 
 
 1017 
 19 
 0.059006211180124224 
 
 
 1083 
 16 
 0.049689440993788817 
 
 
 
 5 
 0.015527950310559006 
 
 
 1044 
 1 
 0.003105590062111801 
 
 
 1052 
 1 
 0.003105590062111801

In [79]:
tbl_new_pf3k_metadata.selectnone('HasDuplicate')

+------------+-------------+--------+----------------------------------------------------------------+----------------------+------------------------+---------+-----------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| sample     | acc         | study  | study_title                                                    | contact_name         | contact_email          | country | site      | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual | DiscardAsDuplicate | HasDuplicate | Individual_ID |
+============+=============+========+================================================================+======================+========================+=========+===========+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+====================+==============+===============+
| 'PF0096-C' | 'ERS013064' | '1017' | 'Population genetics of natural populations in Northern Ghana' | 'Lucas Amenga-Etego' | 'lucasmenga@gmail.com' | 'Ghana' | 'Kassena' | ''              | ''          | '6435175604' | '6267496956' | '1724521900'     | '75'            | '22.9'            | '23296644'           | '23266881'           | '23239650'            | '22761256'            | '251.14'      | '448.9'            | '864.8'          | '89.64'   | True          | False           | 'PF0096-C'               | None               | None         | 'PFD41'       |
+------------+-------------+--------+----------------------------------------------------------------+----------------------+------------------------+---------+-----------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| 'PF0132-C' | 'ERS013078' | '1017' | 'Population genetics of natural populations in Northern Ghana' | 'Lucas Amenga-Etego' | 'lucasmenga@gmail.com' | 'Ghana' | 'Kassena' | ''              | ''          | '4662197809' | '4510803757' | '255092014'      | '75'            | '26'              | '23319493'           | '23293174'           | '23259375'            | '22941906'            | '192.31'      | '338.3'            | '50.3'           | '90.94'   | True          | False           | 'PF0132-C'               | None               | None         | 'PFD140'      |
+------------+-------------+--------+----------------------------------------------------------------+----------------------+------------------------+---------+-----------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| 'PF0133-C' | 'ERS013079' | '1

In [81]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'PFD140')

+------------+-------------+--------+----------------------------------------------------------------+----------------------+------------------------+---------+-----------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| sample     | acc         | study  | study_title                                                    | contact_name         | contact_email          | country | site      | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual | DiscardAsDuplicate | HasDuplicate | Individual_ID |
+============+=============+========+================================================================+======================+========================+=========+===========+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+====================+==============+===============+
| 'PF0132-C' | 'ERS013078' | '1017' | 'Population genetics of natural populations in Northern Ghana' | 'Lucas Amenga-Etego' | 'lucasmenga@gmail.com' | 'Ghana' | 'Kassena' | ''              | ''          | '4662197809' | '4510803757' | '255092014'      | '75'            | '26'              | '23319493'           | '23293174'           | '23259375'            | '22941906'            | '192.31'      | '338.3'            | '50.3'           | '90.94'   | True          | False           | 'PF0132-C'               | None               | None         | 'PFD140'      |
+------------+-------------+--------+----------------------------------------------------------------+----------------------+------------------------+---------+-----------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+

In [66]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', '')

+--------+-----+-------+-------------+--------------+---------------+---------+------+-----------------+-------------+-------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| sample | acc | study | study_title | contact_name | contact_email | country | site | collection_date | sample_prep | bases | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual | DiscardAsDuplicate | HasDuplicate | Individual_ID |
+========+=====+=======+=============+==============+===============+=========+======+=================+=============+=======+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+====================+==============+===============+

In [67]:
tbl_new_pf3k_metadata.selectnone('Individual_ID')

+--------+-----+-------+-------------+--------------+---------------+---------+------+-----------------+-------------+-------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+--------------------------+--------------------+--------------+---------------+
| sample | acc | study | study_title | contact_name | contact_email | country | site | collection_date | sample_prep | bases | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | IsFieldSample | PreferredSample | AllSamplesThisIndividual | DiscardAsDuplicate | HasDuplicate | Individual_ID |
+========+=====+=======+=============+==============+===============+=========+======+=================+=============+=======+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+==========================+====================+==============+===============+

In [69]:
len(tbl_new_pf3k_metadata.selecteq('study', 'Broad Senegal').data())

137

In [46]:
print(len(tbl_new_pf3k_metadata.distinct('AllSamplesThisIndividual').data()))
print(len(tbl_new_pf3k_metadata.distinct(('study', 'AllSamplesThisIndividual')).data()))


2513
2514


In [27]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'HB3')

+-------------+-------------+--------+-------------------------+---------------------------+-----------------------------------------+---------+------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+---------------+-----------------+-------------------+--------------+--------------------+-----------------+-------------------------+--------------------+--------------------------+---------------+---------------+--------------------------+
| sample      | acc         | study  | study_title             | contact_name              | contact_email                           | country | site | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | OxfordSrcCode | OxfordDonorCode | Individual_ID | IndividualGroup | AlfrescoStudyCode | HasDuplicate | DiscardAsDuplicate | pc_pass_missing | pc_genome_covered_at_1x | oxford_source_code | oxford_donor_source_code | crosses_study | crosses_clone | AllSamplesThisIndividual |
+=============+=============+========+=========================+===========================+=========================================+=========+======+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+===============+=================+===================+==============+====================+=================+=========================+====================+==========================+===============+===============+==========================+
| 'PG0004-CW' | 'ERS010539' | '1041' | 'HB3xDD2 cross progeny' | 'Michael Ferdig'          | 'mferdig@nd.edu'                        | ''      | ''   | ''              | ''          | '2071343736' | '1959788960' | '82692599'       | '75'            | '25.5'            | '23028392'           | '22860658'           | '22714261'            | '20225218'            | '84.62'       | '223.1'            | '39.7'           | '89.43'   | None          | None            | 'HB3'         | None            | None              | None         | None               | None            | None                    | 'HB3_Ferdig'       | 'HB3'                    | '1041'        | 'HB3'         | 'PG0004-CW,PG0052-C'     |
+-------------+-------------+--------+-------------------------+---------------------------+-----------------------------------------+---------+------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+---------------+-----------------+-------------------+--------------+--------------------+-----------------+-------------------------+--------------------+--------------------------+---------------+---------------+--------------------------+
| 'PG0052-C'  | 'ERS010000' | '1042' | '3D7xHB3 cross progeny' | 'Lisa Ranford-Cartwright' | 'Lisa.Ranford-Cartwright@glasgow.ac.uk' | ''      | ''   | ''              | ''          | '2500934542' | '2432094350' | '112742917'      | '75'            | '26.6'            | '23020598'           | '22830696'           | '22553002'            | '19003457'            | '104.67'      | '279.5'            | '46.3'           | '89.25'   | None          | None       

In [28]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'C02')

+------------+-------------+--------+-------------------------+---------------------------+-----------------------------------------+---------+------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+---------------+-----------------+-------------------+--------------+--------------------+-----------------+-------------------------+--------------------+--------------------------+---------------+---------------+---------------------------------------+
| sample     | acc         | study  | study_title             | contact_name              | contact_email                           | country | site | collection_date | sample_prep | bases        | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | OxfordSrcCode | OxfordDonorCode | Individual_ID | IndividualGroup | AlfrescoStudyCode | HasDuplicate | DiscardAsDuplicate | pc_pass_missing | pc_genome_covered_at_1x | oxford_source_code | oxford_donor_source_code | crosses_study | crosses_clone | AllSamplesThisIndividual              |
+============+=============+========+=========================+===========================+=========================================+=========+======+=================+=============+==============+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+===============+=================+===================+==============+====================+=================+=========================+====================+==========================+===============+===============+=======================================+
| 'PG0053-C' | 'ERS010012' | '1042' | '3D7xHB3 cross progeny' | 'Lisa Ranford-Cartwright' | 'Lisa.Ranford-Cartwright@glasgow.ac.uk' | ''      | ''   | ''              | ''          | '1855569209' | '1768753573' | '226779855'      | '75'            | '28.5'            | '23141515'           | '23011404'           | '22877922'            | '17537678'            | '75.78'       | '336.8'            | '72.6'           | '91.35'   | None          | None            | 'C02'         | None            | None              | None         | None               | None            | None                    | 'XP3'              | 'XP3'                    | '1042'        | 'C02'         | 'PG0053-C,PG0055-C,PG0056-C,PG0067-C' |
+------------+-------------+--------+-------------------------+---------------------------+-----------------------------------------+---------+------+-----------------+-------------+--------------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+---------------+-----------------+-------------------+--------------+--------------------+-----------------+-------------------------+--------------------+--------------------------+---------------+---------------+---------------------------------------+
| 'PG0055-C' | 'ERS010011' | '1042' | '3D7xHB3 cross progeny' | 'Lisa Ranford-Cartwright' | 'Lisa.Ranford-Cartwright@glasgow.ac.uk' | ''      | ''   | ''              | ''          | '2489898665' | '2432833837' | '86561664'       | '75'            | '29.7'            | '23218176'           | '23095177'           | '22965215'            | '21659942'            | '104.24'      | '282.3'            |

In [29]:
tbl_new_pf3k_metadata.selecteq('Individual_ID', 'A4')

+--------+-----+-------+-------------+--------------+---------------+---------+------+-----------------+-------------+-------+--------------+------------------+-----------------+-------------------+----------------------+----------------------+-----------------------+-----------------------+---------------+--------------------+------------------+-----------+---------------+-----------------+---------------+-----------------+-------------------+--------------+--------------------+-----------------+-------------------------+--------------------+--------------------------+---------------+---------------+--------------------------+
| sample | acc | study | study_title | contact_name | contact_email | country | site | collection_date | sample_prep | bases | bases_mapped | bases_duplicated | avg_read_length | mean_base_quality | bases_of_1X_coverage | bases_of_5X_coverage | bases_of_10X_coverage | bases_of_50X_coverage | mean_coverage | mean_fragment_size | sd_fragment_size | %callable | OxfordSrcCode | OxfordDonorCode | Individual_ID | IndividualGroup | AlfrescoStudyCode | HasDuplicate | DiscardAsDuplicate | pc_pass_missing | pc_genome_covered_at_1x | oxford_source_code | oxford_donor_source_code | crosses_study | crosses_clone | AllSamplesThisIndividual |
+========+=====+=======+=============+==============+===============+=========+======+=================+=============+=======+==============+==================+=================+===================+======================+======================+=======================+=======================+===============+====================+==================+===========+===============+=================+===============+=================+===================+==============+====================+=================+=========================+====================+==========================+===============+===============+==========================+

# Earlier work to determine duplicates
Note this is no longer relevant after final file created

In [12]:
tbl_new_pf3k_metadata.valuecounts('study').displayall()

study 
 count 
 frequency 
 
 
 
 
 1052 
 947 
 0.3587121212121212 
 
 
 1017 
 409 
 0.15492424242424244 
 
 
 1022 
 369 
 0.13977272727272727 
 
 
 1044 
 181 
 0.06856060606060606 
 
 
 Broad Senegal 
 137 
 0.051893939393939395 
 
 
 1083 
 116 
 0.04393939393939394 
 
 
 1026 
 100 
 0.03787878787878788 
 
 
 1001 
 96 
 0.03636363636363636 
 
 
 1094 
 92 
 0.03484848484848485 
 
 
 1006 
 65 
 0.02462121212121212 
 
 
 1043 
 38 
 0.014393939393939395 
 
 
 1041 
 37 
 0.014015151515151515 
 
 
 1104 
 27 
 0.010227272727272727 
 
 
 1042 
 21 
 0.007954545454545454 
 
 
 
 5 
 0.001893939393939394

In [13]:
(
    tbl_new_pf3k_metadata
    .selectnone('Individual_ID')
    .valuecounts('study')
    .displayall()
)

study 
 count 
 frequency 
 
 
 
 
 Broad Senegal 
 137 
 0.4254658385093168 
 
 
 1043 
 38 
 0.11801242236024845 
 
 
 1041 
 37 
 0.11490683229813664 
 
 
 1104 
 27 
 0.08385093167701864 
 
 
 1042 
 21 
 0.06521739130434782 
 
 
 1022 
 20 
 0.062111801242236024 
 
 
 1017 
 19 
 0.059006211180124224 
 
 
 1083 
 16 
 0.049689440993788817 
 
 
 
 5 
 0.015527950310559006 
 
 
 1052 
 1 
 0.003105590062111801 
 
 
 1044 
 1 
 0.003105590062111801

In [23]:
(
    tbl_new_pf3k_metadata
    .selectnone('Individual_ID')
    .valuecounts('study')
    .displayall()
)

study 
 count 
 frequency

In [24]:
(
    tbl_new_pf3k_metadata
    .selecteq('Individual_ID', '')
    .valuecounts('study')
    .displayall()
)

study 
 count 
 frequency

In [14]:
(
    tbl_new_pf3k_metadata
    .selectnotin('study', ['Broad Senegal', '', '1041', '1042', '1043', '1104'])
    .selecteq('AlfrescoStudyCode', None)
    .valuecounts('study', 'AlfrescoStudyCode')
    .displayall()
)

study 
 AlfrescoStudyCode 
 count 
 frequency 
 
 
 
 
 1022 
 None 
 20 
 0.3508771929824561 
 
 
 1017 
 None 
 19 
 0.3333333333333333 
 
 
 1083 
 None 
 16 
 0.2807017543859649 
 
 
 1052 
 None 
 1 
 0.017543859649122806 
 
 
 1044 
 None 
 1 
 0.017543859649122806

In [28]:
(
    tbl_new_pf3k_metadata
    .selectnotin('study', ['Broad Senegal', '', '1041', '1042', '1043', '1104'])
    .selecteq('AlfrescoStudyCode', None)
    .displayall()
)

sample 
 acc 
 study 
 study_title 
 contact_name 
 contact_email 
 country 
 site 
 collection_date 
 sample_prep 
 bases 
 bases_mapped 
 bases_duplicated 
 avg_read_length 
 mean_base_quality 
 bases_of_1X_coverage 
 bases_of_5X_coverage 
 bases_of_10X_coverage 
 bases_of_50X_coverage 
 mean_coverage 
 mean_fragment_size 
 sd_fragment_size 
 %callable 
 OxfordSrcCode 
 OxfordDonorCode 
 Individual_ID 
 IndividualGroup 
 AlfrescoStudyCode 
 HasDuplicate 
 DiscardAsDuplicate 
 pc_pass_missing 
 pc_genome_covered_at_1x 
 oxford_source_code 
 oxford_donor_source_code 
 
 
 
 
 PF0096-C 
 ERS013064 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 6435175604 
 6267496956 
 1724521900 
 75 
 22.9 
 23296644 
 23266881 
 23239650 
 22761256 
 251.14 
 448.9 
 864.8 
 89.64 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD41 
 PFD41 
 
 
 PF0132-C 
 ERS013078 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4662197809 
 4510803757 
 255092014 
 75 
 26 
 23319493 
 23293174 
 23259375 
 22941906 
 192.31 
 338.3 
 50.3 
 90.94 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD140 
 PFD140 
 
 
 PF0133-C 
 ERS013079 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 2777710622 
 2704426026 
 337925397 
 75 
 25.6 
 23275596 
 23183228 
 23088409 
 22218115 
 115.4 
 316.1 
 44.3 
 90.03 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD141 
 PFD141 
 
 
 PF0134-C 
 ERS013080 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 3792406304 
 3651912704 
 513817488 
 75 
 23.2 
 22058430 
 19399764 
 17706152 
 13304906 
 143.51 
 363.3 
 120.8 
 73.56 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD142 
 PFD142 
 
 
 PF0155-C 
 ERS157452 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4851737429 
 4763266529 
 253053064 
 99 
 25.3 
 23323254 
 23308190 
 23280458 
 22059775 
 202.39 
 234.8 
 76.8 
 91.31 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD163 
 PFD163 
 
 
 PF0168-C 
 ERS157456 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4143769367 
 4059828767 
 215410448 
 99 
 25 
 23290843 
 23178742 
 23002812 
 20766485 
 171.54 
 236.4 
 80.1 
 90.58 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD503 
 PFD503 
 
 
 PF0208-C 
 ERS022751 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 12261009471 
 11754929346 
 756068524 
 74 
 24.5 
 23298314 
 23265132 
 23233513 
 22905439 
 453.51 
 292.8 
 181.3 
 89.98 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD215 
 PFD215 
 
 
 PF0222-C 
 ERS022961 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 1407275315 
 1337689190 
 55921222 
 74 
 25.8 
 23206487 
 23033033 
 22885659 
 16178776 
 57.63 
 316 
 96 
 89.94 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD516 
 PFD516 
 
 
 PF0228-C 
 ERS022754 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 7023417574 
 6897793174 
 360148407 
 74 
 25.5 
 23322082 
 23307388 
 23293695 
 23170769 
 292.87 
 283 
 79.4 
 90.99 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD527 
 PFD527 
 
 
 PF0234-C 
 ERS017395 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Luc

In [28]:
(
    tbl_new_pf3k_metadata
    .selectnone('Individual_ID')
    .selecteq('AlfrescoStudyCode', None)
    .displayall()
)

sample 
 acc 
 study 
 study_title 
 contact_name 
 contact_email 
 country 
 site 
 collection_date 
 sample_prep 
 bases 
 bases_mapped 
 bases_duplicated 
 avg_read_length 
 mean_base_quality 
 bases_of_1X_coverage 
 bases_of_5X_coverage 
 bases_of_10X_coverage 
 bases_of_50X_coverage 
 mean_coverage 
 mean_fragment_size 
 sd_fragment_size 
 %callable 
 OxfordSrcCode 
 OxfordDonorCode 
 Individual_ID 
 IndividualGroup 
 AlfrescoStudyCode 
 HasDuplicate 
 DiscardAsDuplicate 
 pc_pass_missing 
 pc_genome_covered_at_1x 
 oxford_source_code 
 oxford_donor_source_code 
 
 
 
 
 PF0096-C 
 ERS013064 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 6435175604 
 6267496956 
 1724521900 
 75 
 22.9 
 23296644 
 23266881 
 23239650 
 22761256 
 251.14 
 448.9 
 864.8 
 89.64 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD41 
 PFD41 
 
 
 PF0132-C 
 ERS013078 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4662197809 
 4510803757 
 255092014 
 75 
 26 
 23319493 
 23293174 
 23259375 
 22941906 
 192.31 
 338.3 
 50.3 
 90.94 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD140 
 PFD140 
 
 
 PF0133-C 
 ERS013079 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 2777710622 
 2704426026 
 337925397 
 75 
 25.6 
 23275596 
 23183228 
 23088409 
 22218115 
 115.4 
 316.1 
 44.3 
 90.03 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD141 
 PFD141 
 
 
 PF0134-C 
 ERS013080 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 3792406304 
 3651912704 
 513817488 
 75 
 23.2 
 22058430 
 19399764 
 17706152 
 13304906 
 143.51 
 363.3 
 120.8 
 73.56 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD142 
 PFD142 
 
 
 PF0155-C 
 ERS157452 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4851737429 
 4763266529 
 253053064 
 99 
 25.3 
 23323254 
 23308190 
 23280458 
 22059775 
 202.39 
 234.8 
 76.8 
 91.31 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD163 
 PFD163 
 
 
 PF0168-C 
 ERS157456 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 4143769367 
 4059828767 
 215410448 
 99 
 25 
 23290843 
 23178742 
 23002812 
 20766485 
 171.54 
 236.4 
 80.1 
 90.58 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD503 
 PFD503 
 
 
 PF0208-C 
 ERS022751 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 12261009471 
 11754929346 
 756068524 
 74 
 24.5 
 23298314 
 23265132 
 23233513 
 22905439 
 453.51 
 292.8 
 181.3 
 89.98 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD215 
 PFD215 
 
 
 PF0222-C 
 ERS022961 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 1407275315 
 1337689190 
 55921222 
 74 
 25.8 
 23206487 
 23033033 
 22885659 
 16178776 
 57.63 
 316 
 96 
 89.94 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD516 
 PFD516 
 
 
 PF0228-C 
 ERS022754 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Lucas Amenga-Etego 
 lucasmenga@gmail.com 
 Ghana 
 Kassena 
 
 
 7023417574 
 6897793174 
 360148407 
 74 
 25.5 
 23322082 
 23307388 
 23293695 
 23170769 
 292.87 
 283 
 79.4 
 90.99 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 None 
 PFD527 
 PFD527 
 
 
 PF0234-C 
 ERS017395 
 1017 
 Population genetics of natural populations in Northern Ghana 
 Luc

# Manual checking
After manual checking, all samples from study 1017 are unique.

1052 has duplicates, but PH0979-C/KH002-110 is unique

For 1022: H183, H185, H196, H198, H220, H224, H229, H237, H250 have duplicates. All can be matched on source code

1083 contains duplicates that can be matched on source code

1044 has one duplicate that needs manual matching